### PR TITLE
Fix managed label conditions and docs

### PR DESCRIPTION
**Pull request by adrianludwin**
Add a very minor fix to ensure that out-of-date managed label conditions
are cleared. Also update a variety of docs.

Tested: existing tests only

### CODE REVIEW

**Comment by adrianludwin** [comment]
/cc @erikgb 
/assign @rjbez17 

wdyt of these doc updates? I know they're a bit sparse but they can be improved in the future.

**Comment by adrianludwin** [comment]
To see the rendered markdown:
* https://github.com/adrianludwin/hierarchical-namespaces/blob/prop-doc/docs/user-guide/how-to.md#use-managed-labels
* https://github.com/adrianludwin/hierarchical-namespaces/blob/prop-doc/docs/user-guide/how-to.md#admin-managed-labels
* https://github.com/adrianludwin/hierarchical-namespaces/blob/prop-doc/docs/user-guide/concepts.md#admin-managed-labels

File: docs/user-guide/concepts.md [hunk 1]
```diff
@@ -505,6 +500,40 @@ can either query such objects directly, or via `kubectl hns describe NAMESPACE`.
...
+
+***Managed labels and annotations are planned for HNC v1.0+***
+
+Just as certain objects can be propagated from parent namespaces to their
+descendants, so can certain labels and annotations on namespaces. For example,
+an admin may define an `environment:prod` label on a parent namespace, and
+ensure that it will be automatically propagated to all descendants of that
+namespace.
+
+However, managed labels (and annotations - the remainer of this section applies
```
Comment (rjbez17) [inline docs/user-guide/concepts.md:515]: nit: remainder

**Review by rjbez17 — COMMENTED** [review]
One nut but lgtm

/lgtm
/hold

File: docs/user-guide/concepts.md [hunk 2]
```diff
@@ -505,6 +500,40 @@ can either query such objects directly, or via `kubectl hns describe NAMESPACE`.
...
 
+<a name="admin-managed-labels"/>
+
+### Managed labels and annotations
+
+***Managed labels and annotations are planned for HNC v1.0+***
+
+Just as certain objects can be propagated from parent namespaces to their
+descendants, so can certain labels and annotations on namespaces. For example,
+an admin may define an `environment:prod` label on a parent namespace, and
```
Comment (erikgb) [inline docs/user-guide/concepts.md:511]: Suggestion: Use a prefixed label in this example. K8s recommends prefixed labels, ref. https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set

File: docs/user-guide/concepts.md (hunk 1 above)
Comment (adrianludwin) [reply docs/user-guide/concepts.md]: Fixed

File: docs/user-guide/concepts.md (hunk 2 above)
Comment (adrianludwin) [reply docs/user-guide/concepts.md]: Done, thanks

**Comment by adrianludwin** [comment]
@rjbez17 May I get a re-lgtm? Thanks!

**Pull request merged**